codes/c/chapter_hashing/array_hash_map.c

```diff
@@ -0,0 +1,174 @@
+/**
+ * File: array_hash_map.c
+ * Created Time: 2023-03-18
+ * Author: Guanngxu (446678850@qq.com)
+ */
+
+#include "../include/include.h"
+
+/* 哈希表初始化函数 */
+array_hash_map_t *new_array_hash_map() {
+    array_hash_map_t *array_hash_map = malloc(sizeof(array_hash_map_t));
+    return array_hash_map;
+}
+
+/* 哈希函数 */
+int hash_func(int key) {
+    int index = key % HASH_MAP_DEFAULT_SIZE;
+    return index;
+}
+
+/* 查询操作 */
+const char *get(const array_hash_map_t *d, const int key) {
+    int index = hash_func(key);
+    const entry_t *pair = d->buckets + index;
+    if(pair == NULL) 
+        return NULL;
+    return pair->val;
+}
+
+/* 添加操作 */
+void put(array_hash_map_t *d, const int key, const char *val) {
+    int index = hash_func(key);
+    d->buckets[index].key = key;
+    d->buckets[index].val = malloc(strlen(val) + 1);
```

Comment:
After modifying the storage array to a pointer, a new `entry` can be created here and linked to `buckets`.

Reply:
[See cpp code for details](https://github.com/krahets/hello-algo/blob/28fdd26f2fcd4caac2c8ce631079492bdce37a95/codes/cpp/chapter_hashing/array_hash_map.cpp#L55)